params.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"regexp"
+	"strings"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+// Params holds the parameter mappings specified  by user
+type Params map[PName]PValue
+type PName string
+type PValue string
+
+type ParamFile struct {
```

Comment:
@hsanjuan Fixed this (went with your second proposal for a more general config) and pushed. Haven't written the tests you mentioned below yet.